File: 백준/Silver/1149. RGB거리/RGB거리.py
import sys
n = int(sys.stdin.readline())
R=[]
for i in range(n):
    r,g,b = map(int,sys.stdin.readline().strip().split()) #strip왼쪽 공백 제거

    if(i==0):
        R=[r,g,b]
        continue
    R = [min(R[1], R[2]) + r, 
           min(R[0], R[2]) + g,
           min(R[0], R[1]) + b]
    
print(min(R))

File: 백준/Silver/1149. RGB거리/RGB거리.js
const fs = require("fs");
// TODO: 제출 시 경로 변환 필수 ("/dev/stdin")
//6.53
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const n = Number(input.shift());
const values = input.map(e => e.split(" ").map(e => Number(e)));

dp = Array.from(Array(n), () => new Array(3).fill(0));

for (let i = 0; i < n; i++) {
  if (i === 0) {
    dp[0][0] = values[0][0];
    dp[0][1] = values[0][1];
    dp[0][2] = values[0][2];
    continue;
  }
  dp[i][0] = Math.min(dp[i - 1][1] + values[i][0], dp[i - 1][2] + values[i][0]);
  dp[i][1] = Math.min(dp[i - 1][0] + values[i][1], dp[i - 1][2] + values[i][1]);
  dp[i][2] = Math.min(dp[i - 1][0] + values[i][2], dp[i - 1][1] + values[i][2]);
}

console.log(Math.min(...dp[n - 1]));
